### PR TITLE
Adds staticcheck and fixes lint errors

### PR DESCRIPTION
**Pull request by markylaing**
Continuing with #10392, this PR includes staticcheck in the static analysis script and fixes all associated lint errors.

Additionally, `govet` and `ineffassign` are added to the static analysis script (there were no lint errors for these).

### CODE REVIEW

File: lxd/devlxd_test.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ type DevLxdDialer struct {
 	Path string
 }
 
-func (d DevLxdDialer) DevLxdDial(network, path string) (net.Conn, error) {
+func (d DevLxdDialer) DevLxdDial(context context.Context, network, path string) (net.Conn, error) {
```
Comment (monstermunchkin) [inline lxd/devlxd_test.go:26]: This should be `ctx` to be consistent.

File: lxd/network/openvswitch/ovn.go [hunk 2]
```diff
@@ -432,7 +432,7 @@ func (o *OVN) LogicalRouterPortAdd(routerName OVNRouter, portName OVNRouterPort,
 
 			_, err := o.nbctl("set", "Logical_Router_Port", string(portName),
 				fmt.Sprintf(`networks="%s"`, strings.Join(ips, `","`)),
-				fmt.Sprintf(`mac="%s"`, fmt.Sprintf(mac.String())),
+				fmt.Sprintf(`mac="%s"`, mac.String()),
```
Comment (monstermunchkin) [inline lxd/network/openvswitch/ovn.go:435]: Could be written as `fmt.Sprintf("mac=%q", mac.String())`.

**Review by monstermunchkin — CHANGES_REQUESTED** [review]

File: lxd/devlxd_test.go (hunk 1 above)
Comment (tomponline) [reply lxd/devlxd_test.go:26]: Yep good spot!

**Review by tomponline — COMMENTED** [review]
As @monstermunchkin mentioned its convention to use ctx for context argument.

File: lxc/remote.go [hunk 3]
```diff
@@ -170,7 +170,7 @@ func (c *cmdRemoteAdd) RunToken(server string, token string, rawToken *api.Certi
 
 		conf.Remotes[server] = config.Remote{Addr: addr, Protocol: c.flagProtocol, AuthType: c.flagAuthType, Domain: c.flagDomain}
 
-		d, err = conf.GetInstanceServer(server)
+		_, err = conf.GetInstanceServer(server)
```
Comment (tomponline) [inline lxc/remote.go:173]: If d isn't used outside of the error check can we move the definition of the var inside too.

Comment (markylaing) [reply lxc/remote.go:172]: Looks like `d` is used outside of the error check but not outside of the for loop so I've removed the definition at the function top level and used a `:=` inside the loop.

**Comment by markylaing** [comment]
> Some commits reference someone called "DailContext" :)

Whoops! Fixed now :smile:

File: lxd/network/driver_ovn.go [hunk 4]
```diff
@@ -4365,7 +4365,8 @@ func (n *ovn) PeerCreate(peer api.NetworkPeersPost) error {
 	// Check if there is an existing peer using the same name, or whether there is already a peering (in any
 	// state) to the target network.
 	peers, err := n.state.DB.Cluster.GetNetworkPeers(n.ID())
-	if err == nil {
+	if err != nil {
+		return api.StatusErrorf(http.StatusServiceUnavailable, "Could not validate against existing peers")
```
Comment (tomponline) [inline lxd/network/driver_ovn.go:4369]: Oh whoops thats one of mine :(

I think we should use http.StatusInternalServerError here (or just return the error, which will also default to internal server error status code) as we use that everywhere else for DB errors.

File: lxd/storage/drivers/utils_ceph.go [hunk 5]
```diff
@@ -110,7 +110,7 @@ func CephMonitors(cluster string) ([]string, error) {
 					server = strings.Split(server, "/")[0]
 
 					// Handle end of nested blocks.
-					server = strings.Replace(server, "]]", "]", 0)
+					server = strings.Replace(server, "]]", "]", -1)
```
[the remark this answers is not in the document]
Comment (markylaing) [reply lxd/storage/drivers/utils_ceph.go]: Changed.

**Pull request merged**